go/runtime/txpool/queues.go

```diff
@@ -0,0 +1,36 @@
+package txpool
+
+import (
+	"time"
+
+	"github.com/oasisprotocol/oasis-core/go/common/crypto/hash"
+)
+
+// UsableTransactionSource is a place to retrieve txs that are "good enough." "Good enough" variously means CheckTx'd,
+// came from roothash incoming message, or came from our own node.
+type UsableTransactionSource interface {
+	// GetSchedulingSuggestion returns some number of txs to give to the scheduler as part of the initial
```

Comment:
Shouldn't the number of transactions be a parameter (e.g. currently the initial batch size comes from the runtime's configuration)?

Reply:
transferring comment from other thread:

> Where did support for iterating over the prioritized suggested batch via offset/limit go? This is needed as we don't want to send the entire transaction queue into the runtime.

should have those parameters probably

Reply:
parameters added.

access to later "pages" of transactions added back too. for now it consults only the main queue. future work can add a request option to specify otherwise